ci/travis/install-dependencies.sh

```diff
@@ -1,122 +1,173 @@
+WORKSPACE_DIR="${ROOT_DIR}/../.."
+
+pkg_install() {
+  case "${OSTYPE}" in
+    darwin*) brew install "$@";;
+    linux*) sudo apt-get install -qq -o=Dpkg::Use-Pty=0 "$@" | grep --line-buffered -v "^\(Preparing to unpack\|Unpacking\|Processing triggers for\) ";;
+    *) false;;
+  esac
+}
+
+install_base() {
+  case "${OSTYPE}" in
+    linux*)
+      sudo apt-get update -qq
+      pkg_install build-essential curl unzip tmux gdb libunwind-dev python3-pip python3-setuptools
+      if [ "${LINUX_WHEELS-}" = 1 ]; then
+        pkg_install docker
+        sudo usermod -a -G docker travis
+      fi
+      if [ -n "${PYTHON-}" ]; then
+        "${ROOT_DIR}/install-strace.sh" || true
+      fi
+      ;;
+  esac
+}
+
+install_miniconda() {
+  local miniconda_version="3-4.5.4" miniconda_platform="" exe_suffix=".sh"
+  case "${OSTYPE}" in
+    linux*) miniconda_platform=Linux;;
+    darwin*) miniconda_platform=MacOSX;;
+    msys*) miniconda_platform=Windows; exe_suffix=".exe";;
+  esac
+  local miniconda_url="https://repo.continuum.io/miniconda/Miniconda${miniconda_version}-${miniconda_platform}-${HOSTTYPE}${exe_suffix}"
+  local miniconda_target="./${miniconda_url##*/}"
+  local miniconda_dir="$HOME/miniconda"
+  curl -s -L -o "${miniconda_target}" "${miniconda_url}"
+  chmod +x "${miniconda_target}"
+  case "${OSTYPE}" in
+    msys*)
+      miniconda_dir="${miniconda_dir}/bin"  # HACK: Compensate for python.exe being in the installation root on Windows
+      MSYS2_ARG_CONV_EXCL="*" "${miniconda_target}" /S /D="$(cygpath -w -- "${miniconda_dir}")"
+      ;;
+    *)
+      "${miniconda_target}" -b -p "${miniconda_dir}" | grep --line-buffered -v "^\(installing: \|installation finished\.\)"
+      ;;
+  esac
+  { local set_x="${-//[^x]/}"; } 2> /dev/null  # save set -x to suppress noise
+  set +x
+  local source_line='PYTHON3_BIN_PATH=python; PATH="$HOME/miniconda/bin:$PATH"; export PYTHON3_BIN_PATH PATH;'
+  test -f ~/.bashrc && grep -x -q -F "${source_line}" -- ~/.bashrc || echo "${source_line}" >> ~/.bashrc
+  test -z "${set_x}" || set -x  # restore set -x
+  eval "${source_line}"
+  python -m pip install --upgrade --quiet pip
+}
+
+install_nvm() {
+  if [ "${OSTYPE}" = msys ]; then
+      local version="1.1.7" NVM_HOME="${HOME}/.nvm"
+      if [ ! -f "${NVM_HOME}/nvm.sh" ]; then
+        mkdir -p -- "${NVM_HOME}"
+        export NVM_SYMLINK="${PROGRAMFILES}\nodejs"
+        (
+          cd "${NVM_HOME}"
+          local target="./nvm-${version}.zip"
+          curl -s -L -o "${target}" "https://github.com/coreybutler/nvm-windows/releases/download/${version}/nvm-noinstall.zip"
+          unzip -q -- "${target}"
+          rm -f -- "${target}"
+          printf "%s\r\n" "root: $(cygpath -w -- "${NVM_HOME}")" "path: ${NVM_SYMLINK}" > settings.txt
+        )
+        printf "%s\n" "export NVM_HOME=\"$(cygpath -w -- "${NVM_HOME}")\"" 'nvm() { "${NVM_HOME}/nvm.exe" "$@"; }' > "${NVM_HOME}/nvm.sh"
+      fi
+  fi
+}
+
+install_pip() {
```

Comment:
Right but Conda isn't always installed (notice it's only installed under `PYTHON` or `LINT` right now), whereas we need `pip` in _all_ situations, so I can't just rely on Conda. (I think this was the existing behavior. But I think it's good because the installation is slow, too, so I'd rather not install it just for pip.)